classes/CartRule.php

```diff
@@ -319,15 +347,17 @@ public static function getCustomerCartRules(
         $highlight_only = false
     ) {
 
-        if (!CartRule::isFeatureActive()) {
+        if (!CartRule::isFeatureActive()
+            || !CartRule::haveCartRuleToday($id_customer)
+        ) {
             return array();
         }
 
         $sql_part1 = '* FROM `' . _DB_PREFIX_ . 'cart_rule` cr
 				LEFT JOIN `' . _DB_PREFIX_ . 'cart_rule_lang` crl ON (cr.`id_cart_rule` = crl.`id_cart_rule` AND crl.`id_lang` = ' . (int) $id_lang . ')';
 
-        $sql_part2 = ' AND cr.date_from < "' . date('Y-m-d H:i:s') . '"
-				AND cr.date_to > "' . date('Y-m-d H:i:s') . '"
+        $sql_part2 = ' AND cr.date_from < "' . date('Y-m-d H:i:00') . '"
+				AND cr.date_to > "' . date('Y-m-d H:i:00') . '"
```

Comment:
this is wrong if date_to is the end date for the voucher : you need to use :59

Reply:
Agree fixed